hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/Inbox.java

```diff
@@ -53,17 +55,34 @@
      */
     void remove();
 
+    /**
+     * Returns an iterator over the items in the inbox in the order they would
+     * be returned by the {@link #poll()} method.
+     * <p>
+     * The returned iterator doesn't support the {@link Iterator#remove()}
+     * method.
+     */
+    @Override @Nonnull
+    Iterator<Object> iterator();
```

Comment:
To avoid the need to copy the inbox into a list: [here](https://github.com/hazelcast/hazelcast-jet/pull/1813/files#diff-427ced55f675c1e227720703503b626fR137) vs [here](https://github.com/hazelcast/hazelcast-jet/pull/1813/files#diff-427ced55f675c1e227720703503b626fL97)

Reply:
But it's not what it actually replaces; it replaces

```
for (Object item; (item = inbox.poll()) != null; )
```

and we've been using this idiom elsewhere, for example [here](https://github.com/hazelcast/hazelcast-jet/pull/1813/files#diff-df866f8f664e2ce35384d8825df5fab9R96). I don't think the difference is justification enough to burden every `Inbox` implementation with the iterator.

Reply:
I see, the iterator doesn't actually consume the items, it just peeks them.